config/config/src/test/java/io/helidon/config/MpConfigTest.java

```diff
@@ -0,0 +1,103 @@
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.arrayContaining;
+import static org.hamcrest.Matchers.hasSize;
+
+/**
+ * Test MicroProfile config implementation.
+ */
+public class MpConfigTest {
+    private static Config config;
+
+    @BeforeAll
+    static void initClass() {
+        Object helidonConfig = io.helidon.config.Config.builder()
+                .addSource(ConfigSources.classpath("io/helidon/config/application.properties"))
+                .addSource(ConfigSources.create(CollectionsHelper.mapOf("mp-1", "mp-value-1",
+                                                                        "mp-2", "mp-value-2",
+                                                                        "app.storageEnabled", "false",
+                                                                        "mp-array", "a,b,c",
+                                                                        "mp-list.0", "1",
+                                                                        "mp-list.1", "2",
+                                                "mp-list.2", "3")))
+                .disableEnvironmentVariablesSource()
+                .disableSystemPropertiesSource()
+                .build();
+
+        assertThat(helidonConfig, instanceOf(Config.class));
+
+        config = (Config) helidonConfig;
+    }
+
+    @Test
+    void testConfigSources() {
+        Iterable<ConfigSource> configSources = config.getConfigSources();
+        List<ConfigSource> asList = new ArrayList<>();
+        for (ConfigSource configSource : configSources) {
+            asList.add(configSource);
+        }
+
+        assertThat(asList, hasSize(2));
+        assertThat(asList.get(0), instanceOf(ClasspathConfigSource.class));
+        assertThat(asList.get(1), instanceOf(MapConfigSource.class));
+
+        ConfigSource classpath = asList.get(0);
+        assertThat(classpath.getValue("app.storageEnabled"), is("true"));
+
+        ConfigSource map = asList.get(1);
+        assertThat(map.getValue("mp-1"), is("mp-value-1"));
+        assertThat(map.getValue("mp-2"), is("mp-value-2"));
+        assertThat(map.getValue("app.storageEnabled"), is("false"));
+    }
+
+    @Test
+    void testOptionalValue() {
+        assertThat(config.getOptionalValue("app.storageEnabled", Boolean.class), is(Optional.of(true)));
+        assertThat(config.getOptionalValue("mp-1", String.class), is(Optional.of("mp-value-1")));
+    }
+
+    @Test
+    void testStringArray() {
+        String[] values = config.getValue("mp-array", String[].class);
+        assertThat(values, arrayContaining("a", "b", "c"));
+    }
+
+    @Test
+    void testIntArray() {
+        Integer[] values = config.getValue("mp-list", Integer[].class);
+        assertThat(values, arrayContaining(1, 2, 3));
+    }
+
+    // TODO if I use app1.node1.value instead to test overriding, the override fails
```

Comment:
Just noting the TODO

Reply:
Yep, that is a bug. I will create an issue for it.

Reply:
#1182